mne/source_estimate.py

```diff
@@ -1894,18 +1894,29 @@ def save(self, fname, ftype="stc", *, overwrite=False, verbose=None):
             spaces are obtained by adding ``"-lh.stc"`` and ``"-rh.stc"`` (or
             ``"-lh.w"`` and ``"-rh.w"``) to the stem provided, for the left and
             the right hemisphere, respectively.
-        ftype : str
-            File format to use. Allowed values are ``"stc"`` (default),
-            ``"w"``, and ``"h5"``. The ``"w"`` format only supports a single
-            time point.
+        ftype : "auto" | "stc" | "w" | "h5"
+            File format to use. If "auto", the file format will be inferred from the
+            file extension if possible. Other allowed values are ``"stc"``, ``"w"``, and
+            ``"h5"``. The ``"w"`` format only supports a single time point.
         %(overwrite)s
 
             .. versionadded:: 1.0
         %(verbose)s
         """
         fname = str(_check_fname(fname=fname, overwrite=True))  # checked below
+        if ftype == "auto":
+            if fname.endswith((".stc", "-lh.stc", "-rh.stc")):
+                ftype = "stc"
+            elif fname.endswith((".w", "-lh.w", "-rh.w")):
```

Comment:
This can be simplified, as the first element in each `endswith` tuple already captures all cases we care about.